_posts/en/2019-01-20-rbf-in-the-wild.md

```diff
@@ -0,0 +1,242 @@
+
+![rbf-transactions-in-2018](/img/posts/rbf-in-the-wild/rbf-transactions-in-2018.png)
+*5.35% of transactions signal RBF in 2018. Source: [Bitcoin Optech Dashboard](https://dashboard.bitcoinops.org/d/ZsCio4Dmz/rbf-signalling?orgId=1&from=1514835702976&to=1546285302976)*
+
+## What was evaluated?
+
+Wallets were evaluated on the following criteria:
+
+- Can I send an RBF signaled transaction?
+- Can I replace the original transaction?
+- Can I see that an incoming transaction has RBF signaled?
+- Can I see an incoming replacement transaction?
+
+Bitcoin Block Explorers were evaluated on the following criteria:
+
+- Can I see that a transaction has RBF signaled?
+- Can I see the original transaction after it has been replaced?
+
+Additionally, usability, in the form of captioned screenshots, were documented for each criteria.
+
+There are many advanced scenarios which can involve RBF transaction replacements. For purposes of this study, the test only involved single transaction replacements.
+
+During the timeframe of analysis, August - November 2018, the aggregate findings show:
+
+- 6/23 wallets tested support sending RBF transaction in some way
+- 2/23 wallets tested have some indication the transaction being received has RBF signaled
+- 2/12 explorers tested show some indicator that a transaction has RBF signaled
+
+## Some Usability Examples
+
+Below are some select examples of how different services handle RBF from a usability perspective. None of the examples is included to be critical of any particular piece of software, but demonstrating overall how immature RBF’s usability is.
+
+### Electrum: Show RBF Flag in Tx List
+
+![Electrum subtle labeling of RBF transaction screenshot](/img/posts/rbf-in-the-wild/rbf-electrum-label.png)
+
+In this example Electrum tags an RBF signaled transaction with a subtle “rbf” label. Electrum is one of the few wallets that supports RBF for both sending and receiving.
+
+Labeling an unconfirmed transaction in the wallet with RBF visually identifies which outgoing transactions can be fee bumped by the user if there is a delay in confirming. The RBF label also shows which incoming transactions can be fee bumped by their sender using BIP125.
+
+### Bitcoin Core: Increase Transaction Fee, almost
+
+![Bitcoin Bore increase transaction fee failure screenshots](/img/posts/rbf-in-the-wild/rbf-bitcoin-core-increase-fee-fails.png)
+
+In this example Bitcoin Core allows the sending of a transaction with RBF signaled. After the initial transaction broadcast, Bitcoin Core then shows the “Increase transaction fee” option. However, the fee bump fails during the actual bumping of the transaction due to a missing change output. There is a similar user experience if the change address is not large enough to increase the transaction fee.
+
+Both missing change output as well as insufficient change output failures can be resolved by adding additional inputs to the replacement transaction during the bumping.
```

Comment:
Did this bug get reported upstream?  If so, it'd be good to link to it.

Reply:
I did not file an issue for this. I chatted in the irc channel I believe. And I think there was a PR to address this but I didn’t immediately find it. Ill look for it again.

Reply:
I opened an issue regarding this today.